dynamic_db_router/router.py

```diff
@@ -1,29 +1,35 @@
 import threading
 from functools import wraps
 from uuid import uuid4
+from contextvars import ContextVar
 
 from django.db import connections
 
 THREAD_LOCAL = threading.local()
 
+DB_FOR_READ_OVERRIDE = ContextVar('DB_FOR_READ_OVERRIDE', default='default')
```

Comment:
clarification: Does ContextVar still support synchronous threads? Or does this only work for async?

Reply:
From my understanding, contextvar supports synchronous threads. I'll link the documentation but it seems that contextvar behaves similar to threading.local(), which is what was previously used, when values are assigned to different threads. I also tested this out by using the modifed context manager in a synchronous function and everything worked as expected. Here's the link to doc: [https://docs.python.org/3.9/library/contextvars.html]

Reply:
Fantastic. Great work, David!